### PR TITLE
Dcc Server (EXPOSUREAPP-7504)

### PR DESCRIPTION
**Pull request by chris-cwa**
- [x] merge https://github.com/corona-warn-app/cwa-app-android/pull/3332

### CODE REVIEW

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/server/CovidCertificateApiV1.kt [hunk 1]
```diff
@@ -0,0 +1,33 @@
+package de.rki.coronawarnapp.covidcertificate.server
+
+import com.google.gson.annotations.SerializedName
+import retrofit2.Response
+import retrofit2.http.Body
+import retrofit2.http.POST
+
+interface CovidCertificateApiV1 {
```
Comment (chris-cwa) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/server/CovidCertificateApiV1.kt:8]: this is new

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/server/CovidCertificateServer.kt [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+import dagger.Reusable
+import de.rki.coronawarnapp.coronatest.type.RegistrationToken
+import de.rki.coronawarnapp.util.coroutine.DispatcherProvider
+import de.rki.coronawarnapp.util.encryption.rsa.RSAKey
+import kotlinx.coroutines.withContext
+import timber.log.Timber
+import javax.inject.Inject
+
+@Reusable
+class CovidCertificateServer @Inject constructor(
```
Comment (chris-cwa) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/server/CovidCertificateServer.kt:13]: this is implemented

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/server/CovidCertificateServer.kt [hunk 3]
```diff
@@ -1,27 +1,51 @@
...
     suspend fun requestCertificateForTest(
         testRegistrationToken: RegistrationToken,
-    ): TestCertificateComponents {
+    ): TestCertificateComponents = withContext(dispatcherProvider.IO) {
         Timber.tag(TAG).v("requestCertificateForTest(token=%s)", testRegistrationToken)
-        throw NotImplementedError()
+        val response = api.getComponents(
+            requestBody = CovidCertificateApiV1.ComponentsRequest(testRegistrationToken)
+        )
+        // TODO replace with InvalidTestCertificateException + correct error codes
```
Comment (chiljamgossow) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/server/CovidCertificateServer.kt:42]: Will do so with another PR for ticket 7508

**Review by BMItr — APPROVED** [review]
thx for resolving the conflicts chilja!
nothing to complain

**Review by mtwalli — APPROVED** [review]
Lgtm
